### PR TITLE
text validation

### PR DESCRIPTION
**Pull request by hwigmore**
i added a bit of state in the react preview to show off the new validity functionality. might want to look at https://github.com/floored/blocks/pull/17/files?w=1 to see it without whitespace changes

also had to add state back into `TextField.jsx`

i made it so that after your initial invalid message, when you type, we check again so you dont have to do another blur to make that error go away

### CODE REVIEW

**Comment by hwigmore** [comment]
https://app.clubhouse.io/floored/story/9025/validate-text-input-on-blur

File: react/input/TextField.jsx [hunk 1]
```diff
@@ -1,88 +1,99 @@
...
-  let invalidLabelMarker = null;
-  let invalidLabelMessage = null;
-  if (!props.isValid(props.value)) {
-    invalidLabelMarker = (
-      <span className="invalid-label-marker">*</span>
-    );
-    invalidLabelMessage = (
-      <span className="invalid-label-message">{props.invalidErrorMessage}</span>
-    );
+    this.state = { isValid: true };
```
Comment (hwigmore) [inline react/input/TextField.jsx]: oh maybe this shouldn't start as `true` but should start as `isValid: props.isValid(props.value)`... hm

File: react/input/TextField.jsx [hunk 2]
```diff
@@ -1,88 +1,99 @@
...
-  let suffixElement = null;
-  if (props.suffix) {
-    suffixElement = (
-      <span className="text-field-suffix">{props.suffix}</span>
-    );
+    // red asterisk if the value is invalid
+    let invalidLabelMarker = null;
+    let invalidLabelMessage = null;
+    if (!this.state.isValid) {
+      invalidLabelMarker = <span className="invalid-label-marker">*</span>;
```
Comment (hwigmore) [inline react/input/TextField.jsx]: oh i deleted a lot of extra lines just for less scrolling but this is totally personal pref and if you are against it i'll put them back!

**Comment by emma-floored** [comment]
I'm curious about the changes it's showing for `package-lock.json`... do you know what's up with that?

**Comment by hwigmore** [comment]
oh shoot i didnt mean to commit those. although i am curious too. whenever i `npm install` blocks (whether as a suprepo or in its own repo) i get updates there. will check it out.

File: react/input/TextField.jsx [hunk 3]
```diff
@@ -1,88 +1,99 @@
...
-      <span className="invalid-label-marker">*</span>
-    );
-    invalidLabelMessage = (
-      <span className="invalid-label-message">{props.invalidErrorMessage}</span>
-    );
+    this.state = { isValid: true };
+  }
 
-    textFieldClasses += ' invalid';
+  udpateValidity() {
```
Comment (emma-floored) [inline react/input/TextField.jsx]: `update`

File: react/input/TextField.jsx (hunk 2 above)
Comment (emma-floored) [reply react/input/TextField.jsx]: I'm fine with it, I think I mostly just have trouble processing the `>;` combination of characters... it seems crazy

File: react/input/TextField.jsx [hunk 4]
```diff
@@ -1,88 +1,99 @@
...
+            this.props.onBlur(e);
+          }
+        }}
+        onKeyUp={(e) => {
+          if (e.keyCode === 13) { // ENTER
+            e.target.blur();
+          } else if (this.props.onKeyUp) {
+            this.props.onKeyUp(e);
+          }
+          // If an inValid message has already appeared via onBlur,
```
Comment (emma-floored) [inline react/input/TextField.jsx]: I know this is just a comment but the camel case here is kinda funny

File: react/input/TextField.jsx (hunk 1 above)
Comment (MagdalenaHenke) [reply react/input/TextField.jsx]: ☝️ 👏

**Comment by hwigmore** [comment]
ah yeah it was become i was on an older version of node and therefore npm. will get rid of those package-lock changes

**Comment by emma-floored** [comment]
👍

**Pull request merged**